tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+			l := k8s.caLaunch(m, org.Name, caImage)
+			l.Ports = []int32{caPort}
+			launchConfig = append(launchConfig, l)
+			caPort++
+		}
 	}
-	for i := 0; i < len(k.Arguments); i++ {
-		switch k.Action {
-		case "apply", "delete":
-			args = append(args, []string{"-f", k.Arguments[i]}...)
-		default:
-			args = append(args, k.Arguments[i])
+
+	var ordererPort int32 = 30000
+	var ordererMetricsPort int32 = 32500
+	for i := 0; i < len(nsConfig.OrdererOrganizations); i++ {
+		org := nsConfig.OrdererOrganizations[i]
+		for j := 0; j < org.NumOrderers; j++ {
+			err := k8s.generateOrdererConfig(fmt.Sprintf("orderer%d-%s", j, org.Name), org.Name, org.MSPID, ordererPort, ordererMetricsPort, ordererConfig)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to generate orderer configuration file")
+			}
+
+			l := LaunchConfig{Name: fmt.Sprintf("orderer%d-%s", j, org.Name), Type: "orderer"}
+			containers := make([]v12.Container, 0)
+			container := v12.Container{Name: "orderer", Command: []string{"orderer"}, Resources: k8s.resources("orderer", nsConfig.K8s.Resources.Orderers)}
+			container.Image = ordererImage
+			env := v12.EnvVar{Name: "FABRIC_LOGGING_SPEC", Value: nsConfig.OrdererFabricLoggingSpec}
+			container.Env = append(container.Env, env)
+			container.VolumeMounts = k8s.volumeMountLists("orderer", nsConfig.K8s.DataPersistence, nsConfig.EnableNodeOUs)
+			containers = append(containers, container)
+			l.Containers = containers
+			l.Volumes = k8s.volumesList("orderer", org.Name, fmt.Sprintf("orderer%d-%s", j, org.Name), nsConfig.K8s.DataPersistence, nsConfig.EnableNodeOUs)
+			l.Ports = []int32{ordererPort, ordererMetricsPort}
+			launchConfig = append(launchConfig, l)
+			ordererPort++
+			ordererMetricsPort++
+		}
+		for m := 0; m < org.NumCA; m++ {
+			l := k8s.caLaunch(m, org.Name, caImage)
+			l.Ports = []int32{caPort}
+			launchConfig = append(launchConfig, l)
+			caPort++
 		}
 	}
-	return args
+	return launchConfig, nil
 }
 
-func (k K8s) ConfigMapsNSecretsArgs(componentName, k8sType string) []string {
+func (k8s K8s) resources(component string, resource networkspec.Resource) v12.ResourceRequirements {
+	limitsCPU := apiResource.MustParse(resource.Limits.CPU)
+	limitsMemory := apiResource.MustParse(resource.Limits.Memory)
+	requestsCPU := apiResource.MustParse(resource.Requests.CPU)
+	requestsMemory := apiResource.MustParse(resource.Requests.Memory)
+	limitsMap := make(map[v12.ResourceName]apiResource.Quantity)
+	requestsMap := make(map[v12.ResourceName]apiResource.Quantity)
+	limitsMap[v12.ResourceName("cpu")] = apiResource.Quantity(limitsCPU)
+	limitsMap[v12.ResourceName("memory")] = apiResource.Quantity(limitsMemory)
+	requestsMap[v12.ResourceName("cpu")] = apiResource.Quantity(requestsCPU)
+	requestsMap[v12.ResourceName("memory")] = apiResource.Quantity(requestsMemory)
+	resources := v12.ResourceRequirements{Limits: v12.ResourceList(limitsMap), Requests: v12.ResourceList(requestsMap)}
+	return resources
+}
```

Comment:
This can be greatly simplified,

```
func (k8s K8s) resources(resource networkspec.Resource) corev1.ResourceRequirements {
	return corev1.ResourceRequirements{
		Limits: corev1.ResourceList{
			"cpu":    apiResource.MustParse(resource.Limits.CPU),
			"memory": apiResource.MustParse(resource.Limits.Memory),
		},
		Requests: corev1.ResourceList{
			"cpu":    apiResource.MustParse(resource.Requests.CPU),
			"memory": apiResource.MustParse(resource.Requests.Memory),
		},
	}
}
```

Reply:
thank you for this. Fixed